components/org.wso2.carbon.identity.provisioning.connector.google/src/test/java/org/wso2/carbon/identity/provisioning/connector/google/GoogleProvisioningConnectorConfigTest.java

```diff
@@ -0,0 +1,137 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.provisioning.connector.google;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Properties;
+
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.carbon.identity.provisioning.connector.google.GoogleProvisioningConnectorTestConstants
+        .EMAIL_NAME_CLAIM;
+import static org.wso2.carbon.identity.provisioning.connector.google.GoogleProvisioningConnectorTestConstants
+        .LAST_NAME_CLAIM;
+import static org.wso2.carbon.identity.provisioning.connector.google.GoogleProvisioningConnectorTestConstants
+        .REQUIRED_FIELDS;
+import static org.wso2.carbon.identity.provisioning.connector.google.GoogleProvisioningConnectorTestConstants
+        .STREET_ADDRESS_CLAIM;
+
+@PrepareForTest({LogFactory.class})
+public class GoogleProvisioningConnectorConfigTest {
+
+    private static final int CASE_1 = 1;
+    private static final int CASE_2 = 2;
+    private static final int CASE_3 = 3;
+
+    @Mock
+    private Log log;
+
+    @DataProvider(name = "requiredFieldsProvider")
+    public Object[][] provideTestData() {
+
+        Properties properties1 = new Properties();
+        properties1.setProperty(GoogleConnectorConstants.PropertyConfig.REQUIRED_FIELDS, REQUIRED_FIELDS);
+
+        Properties properties2 = new Properties();
+
+        return new Object[][]{
+                {properties1, false},
+                {properties2, true}
+        };
+    }
+
+    @Test(dataProvider = "requiredFieldsProvider")
+    public void testGetRequiredAttributeNames(Properties properties, boolean isEmpty) throws Exception {
+
+        setLogging(false);
+        GoogleProvisioningConnectorConfig connectorConfig = new GoogleProvisioningConnectorConfig(properties);
+
+        if (isEmpty) {
+            Assert.assertEquals(connectorConfig.getRequiredAttributeNames().size(), 0);
+        } else {
+            Assert.assertNotEquals(connectorConfig.getRequiredAttributeNames().size(), 0);
+        }
+    }
+
+    @Test(dataProvider = "requiredFieldsProvider")
+    public void testGetValue(Properties properties, boolean isEmpty) throws Exception {
+
+        setLogging(false);
+        GoogleProvisioningConnectorConfig connectorConfig = new GoogleProvisioningConnectorConfig(properties);
+
+        if (isEmpty) {
```

Comment:
How about passing the expected value from DataProvider and use it to Assert rather than asserting conditionally